src/automations/types/deploymentStatusTrigger.ts

```diff
@@ -0,0 +1,27 @@
+import { isAfterResource, isExpectResource, isForEachResource, isMatchResource } from '@/automations/maps/utilities'
+import { AutomationTriggerEventPosture, DEFAULT_EVENT_TRIGGER_THRESHOLD } from '@/automations/types/automationTriggerEvent'
+import { AutomationTrigger, isAutomationTriggerEvent } from '@/automations/types/triggers'
+import { DeploymentStatus } from '@/models/DeploymentStatus'
+import { createTuple } from '@/utilities/tuples'
+
+export const { values: deploymentStatusEvent, isValue: isDeploymentStatusEvent } = createTuple(['prefect.deployment.ready', 'prefect.deployment.not-ready'])
+export type DeploymentStatusEvent = typeof deploymentStatusEvent[number]
+
+export type DeploymentStatusTrigger = {
+  deployments: string[],
+  posture: AutomationTriggerEventPosture,
+  status: DeploymentStatus,
+  time: number,
+}
+
+/**
+ * This is used to determine if an AutomationTrigger can be mapped into a DeploymentStatusTrigger
+ */
+export function isDeploymentStatusTrigger(trigger: AutomationTrigger): boolean {
+  return isAutomationTriggerEvent(trigger) &&
```

Comment:
Calling out I change line 21. It used to check if the posture was `Proactive` or `Reactive`. But posture is unique to event triggers now, so just checking that its an event trigger.